nemo/deploy/nlp/inference/inference_base.py

```diff
@@ -0,0 +1,420 @@
+
+    Args:
+        checkpoint_path (Union[str, Path]): Path to the NeMo checkpoint directory
+        tensor_model_parallel_size (Optional[int]): Desired tensor-parallel world size (defaults to checkpoint value)
+        pipeline_model_parallel_size (Optional[int]): Desired pipeline-parallel world size (defaults to checkpoint value)
+        context_parallel_size (Optional[int]): Desired context-parallel world size (defaults to checkpoint value)
+        expert_model_parallel_size (Optional[int]): Desired expert parallel world size (defaults to checkpoint value)
+        params_dtype (Optional[torch.dtype]): Data type for model parameters (defaults to checkpoint dtype)
+        micro_batch_size (Optional[int]): Micro batch size for model execution (defaults to 1)
+        enable_flash_decode (bool): Whether to enable flash attention decoding
+        enable_cuda_graphs (bool): Whether to enable CUDA graphs optimization
+
+    Returns:
+        Tuple[List[MegatronModule], MCoreTokenizerWrappper]: Tuple containing:
+            - List of instantiated Megatron-Core modules
+            - Tokenizer wrapper with encode/decode interface
+
+    Raises:
+        ValueError: If checkpoint_path is not a valid NeMo-2.0 checkpoint
+    """
+    checkpoint_path = Path(checkpoint_path)
+
+    # Load model context for config and tokenizer
+    model_context = io.load_context(path=ckpt_to_context_subdir(checkpoint_path), subpath="model")
+
+    model_config = model_context.config
+
+    # Apply ModelOpt specs if they exist in the checkpoint
+    set_modelopt_spec_if_exists_in_ckpt(model_context, checkpoint_path)
+
+    if tensor_model_parallel_size is not None:
+        model_config.tensor_model_parallel_size = tensor_model_parallel_size
+    if pipeline_model_parallel_size is not None:
+        model_config.pipeline_model_parallel_size = pipeline_model_parallel_size
+    if context_parallel_size is not None:
+        model_config.context_parallel_size = context_parallel_size
+    if expert_model_parallel_size is not None:
+        model_config.expert_model_parallel_size = expert_model_parallel_size
+
+    if params_dtype is None:
+        params_dtype = model_config.params_dtype
+
+    if micro_batch_size is None:
+        micro_batch_size = 1
+
+    is_dist_ckpt = check_is_distributed_checkpoint(ckpt_to_weights_subdir(checkpoint_path, is_saving=False))
+    if not is_dist_ckpt:
+        raise ValueError("Checkpoint is not a NeMo-2 distributed checkpoint")
+
+    # Initialize Megatron for inference
+    rng_config = RNGConfig(inference_rng_tracker=True)
+    dist_config = DistributedInitConfig(distributed_backend="nccl")
+    initialize_megatron_for_inference(model_config, dist_config, rng_config, micro_batch_size)
+
+    # Needed for model creation
+    if not model_config.vocab_size:
+        model_config.vocab_size = model_context.tokenizer.vocab_size
+
+    # Enable flash attention
+    if enable_flash_decode:
```

Comment:
I spoke to Helen, who said both of thse can default to True. Tomorrow I will raise an MR on mcore side to set them to True by default. so you can also just do that ?